inject/src/main/kotlin/ktx/inject/inject.kt

```diff
@@ -151,13 +158,23 @@ open class Context : Disposable {
 
   /**
    * Allows to bind a singleton instance to multiple classes in its hierarchy.
-   * @param singleton instance of class compatible with the passed types.
    * @param to list of interfaces and classes in the class hierarchy of the singleton. Any time any of the passed classes
    *    will be requested for injection, the selected singleton will be returned.
+   * @param singleton instance of class compatible with the passed types.
+   * @throws InjectionException if provider for any of the selected types is already defined.
+   */
+  fun <Type : Any> bindSingleton(vararg to: Class<out Type>, singleton: Type)
+          = bind(*to, provider = SingletonProvider(singleton))
+
+  /**
+   * Allows to bind a provider instance to multiple classes in its hierarchy.
+   * @param to list of interfaces and classes in the class hierarchy of the provider. Any time any of the passed classes
+   *    will be requested for injection, the selected provider will be returned.
+   * @param provider provides instances of classes compatible with the passed types.
    * @throws InjectionException if provider for any of the selected types is already defined.
    */
-  fun <Type : Any> bindSingleton(singleton: Type, vararg to: Class<out Type>)
-      = bind(*to, provider = SingletonProvider(singleton))
+  fun <Type : Any> bindSingleton(vararg to: Class<out Type>, provider: () -> Type)
+          = bind(*to, provider = SingletonProvider(provider()))
```

Comment:
Use KTX formatter. We will migrate to official Kotlin formatting style, but I'd like us to stay consistent for now.